src/TSTransformer/nodes/binding/transformArrayAssignmentPattern.ts

```diff
@@ -26,16 +25,23 @@ export function transformArrayAssignmentPattern(
 	for (let element of assignmentPattern.elements) {
 		if (ts.isOmittedExpression(element)) {
 			accessor(state, parentId, index, idStack, true);
-		} else if (ts.isSpreadElement(element)) {
-			DiagnosticService.addDiagnostic(errors.noSpreadDestructuring(element));
 		} else {
 			let initializer: ts.Expression | undefined;
 			if (ts.isBinaryExpression(element)) {
 				initializer = skipDownwards(element.right);
 				element = skipDownwards(element.left);
 			}
 
-			const value = accessor(state, parentId, index, idStack, false);
+			const destructor = ts.isSpreadElement(element)
```

Comment:
I'd suggest moving this variable to a `Lazy` at the top level, next to `accessor`. That way we avoid double work with the typecheck, and can have simpler logic in the element loop

(Also applies to other cases of `destructor` variables)

Reply:
If `value` will end up not being used it'll push an extra id into `idStack`

Reply:
*This comment was meant to be one line lower, referring to the `getSpreadDestructorForType` call